python/ray/dataframe/csv_formatter.py

```diff
@@ -0,0 +1,296 @@
+from __future__ import print_function
+
+import csv as csvlib
+import numpy as np
+import pandas as pd
+import ray
+
+from pandas.core.dtypes.missing import notna
+from pandas.core.index import Index, MultiIndex
+from pandas import compat
+from pandas.compat import (StringIO, range, zip)
+
+from pandas.io.common import (_get_handle, UnicodeWriter, _expand_user,
+                              _stringify_path)
+# from pandas._libs import writers as libwriters
+import pandas._libs.lib as lib
+from pandas.core.indexes.datetimes import DatetimeIndex
+from pandas.core.indexes.period import PeriodIndex
+
+
+class CSVFormatter(object):
```

Comment:
What is the motivation for writing our own `CSVFormatter`?

Reply:
Initial motivation was to write partitions in parallel. I'll see if I can use the Pandas CSVFormatter to write to different offsets of the same file without re-implementing.